### PR TITLE
[DS-3991] made sure that the halbrowser now correctly downloads files

### PR DESCRIPTION
**Pull request by Raf-atmire**
Implemented a fix that allows the HAL browser to properly download files when trying to access the content of a bitstream through the BitstreamContentRestController.

Related jira ticket: https://jira.duraspace.org/browse/DS-3991

### CODE REVIEW

**Comment by tdonohue** [comment]
@Raf-atmire : it looks like this PR was created off a slightly outdated branch. It has merge conflicts.  Could you rebase it against the latest `master` branch, please?

**Comment by Raf-atmire** [comment]
@tdonohue : It was indeed created from an outdated master branch, I've rebased this onto the current master branch now.

**Review by tdonohue — APPROVED** [review]
👍 I've given this a test today.  Before this PR is applied, I can reproduce the download bug (HAL Browser cannot trigger a download, as noted in DS-3991).  After this PR is applied, the HAL Browser works correctly and the file is downloaded.

File: dspace-spring-rest/src/main/webapp/js/hal/http/client.js [hunk 1]
```diff
@@ -7,13 +7,55 @@
...
+    if (authorizationHeader != undefined) {
+        request.setRequestHeader("Authorization", authorizationHeader);
+    }
+    request.onload = function () {
+        // Only handle status code 200
+        if (request.status === 200) {
+            // Try to find out the filename from the content disposition `filename` value
+            var disposition = request.getResponseHeader('content-disposition');
+            var matches = /"([^"]*)"/.exec(disposition);
+            var filename = (matches != null && matches[1] ? matches[1] : 'file.pdf');
```
Comment (tdonohue) [inline dspace-spring-rest/src/main/webapp/js/hal/http/client.js]: One minor comment (via code review).  Is this defaulting the downloaded file to be named "file.pdf"?  If so, I think we should change that behavior, as we should never assume a specific file format (PDF).  If we must have a default name, then I think we should remove the extension here, and simply call it "file"

**Review by tomdesair — APPROVED** [review]
If the remark of @tdonohue is fixed, this PR is OK for me. If the backend does not provide a filename, maybe we should just use `download` or `content` (like the endpoint name).

File: dspace-spring-rest/src/main/webapp/js/hal/http/client.js (hunk 1 above)
Comment (Raf-atmire) [reply dspace-spring-rest/src/main/webapp/js/hal/http/client.js]: This was indeed the case. I've changed the default name to 'content'

**Comment by tdonohue** [comment]
@Raf-atmire : thanks for making that last minor update. This now looks good, and it has two +1 votes.  Merging immediately.

**Pull request merged**